torch/csrc/distributed/c10d/UCCTracing.cpp

```diff
@@ -32,21 +34,21 @@ void ProcessGroupUCCLogger::flushComms(int rank, int world_size) {
         "_", (1 + ltm->tm_mon), "_", ltm->tm_mday, "_", (1900 + ltm->tm_year));
   }
 
-  std::string fullpath = "/tmp/" + dirname;
+  std::filesystem::path fullpath = std::filesystem::path("/tmp") / dirname;
   auto user_path = c10::utils::get_env("TORCH_UCC_COMMS_TRACE_OUTPUT_DIR");
   if (user_path.has_value()) {
     fullpath = std::move(user_path.value());
   }
-  std::string trace_filename = c10::str(fullpath, "/rank", rank, ".json");
-  std::ofstream _outfile;
-  if (!_outfile.is_open()) {
-    if (!mkdir(fullpath.c_str(), 0777)) {
-      LOG(INFO) << getLogPrefix() << "[INFO] failed to mkdir " << fullpath;
-    } else if (errno != EEXIST) {
-      return;
-    }
-    _outfile.open(trace_filename, std::ofstream::out | std::ofstream::trunc);
+  std::filesystem::path trace_filename =
+      fullpath / fmt::format("rank{}.json", rank);
+  std::error_code ec{};
+  if (!std::filesystem::create_directories(fullpath, ec)) {
+    LOG(INFO) << getLogPrefix() << "[INFO] failed to mkdir " << fullpath
+              << " with error " << ec.message();
+    return;
```

Comment:
What happens if error is EExists? We shouldn't return then.

Reply:
create_directories returns true for EExists.